src/Shared/ServerInfrastructure/StringUtilities.cs

```diff
@@ -52,15 +54,13 @@ public static unsafe string GetAsciiOrUTF8StringNonNullCharacters(this ReadOnlyS
             }
         }
 
-        private static readonly SpanAction<char, IntPtr> s_getAsciiOrUtf8StringNonNullCharacters = GetAsciiOrUTF8StringNonNullCharacters;
-
-        private static unsafe void GetAsciiOrUTF8StringNonNullCharacters(Span<char> buffer, IntPtr state)
+        private static unsafe void GetAsciiStringNonNullCharacters(Span<char> buffer, IntPtr state)
```

Comment:
What is this crazy IntPtr argument??

Reply:
This method is the SpanAction passed to [string.Create](https://docs.microsoft.com/en-us/dotnet/api/system.string.create?view=netcore-3.1). Unfortunately, `ReadOnlySpan<byte>` "may not be used as [the TState] type argument" to this method, so it's referenced as an IntPtr [here](https://github.com/dotnet/aspnetcore/blob/5155e11120cf7ee2e07383225057f66512f00fde/src/Shared/ServerInfrastructure/StringUtilities.cs#L30).